From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Solus Linux Team <kernel@getsol.us>
Date: Sat, 5 Jul 2026 00:00:00 +0000
Subject: [PATCH] solus-0002-gaming-latency.patch

Gaming latency reduction patch. Reduces scheduler latency for
gaming workloads by:
  - Reducing sched_min_granularity_ns for faster task switching
  - Reducing sched_wakeup_granularity_ns for faster wakeups
  - Increasing sched_migration_cost_ns to reduce unnecessary migration
  - Tuning sched_latency_ns for lower latency on desktop workloads

Inspired by CachyOS and PDS scheduler tuning.

Signed-off-by: Solus Linux Team <kernel@getsol.us>
---
 kernel/sched/fair.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -60,10 +60,10 @@
  * targeted at desktop + gaming workloads.
  */
-#define sched_latency_ns_default 6000000ULL
-#define sched_min_granularity_ns_default 800000ULL
-#define sched_wakeup_granularity_ns_default 1000000ULL
-#define sched_migration_cost_ns_default 500000ULL
+#define sched_latency_ns_default 4000000ULL
+#define sched_min_granularity_ns_default 500000ULL
+#define sched_wakeup_granularity_ns_default 500000ULL
+#define sched_migration_cost_ns_default 1000000ULL
-- 
2.45.0
